### PR TITLE
feat: dcutr

### PR DESCRIPTION
**Pull request by DannyS03**
## Context

- Adds content for dcutr.
- Part of overall content additions for the NAT section.

## Latest preview

Please view the latest Fleek preview [here](https://bafybeidfzde4b4h32j2h3is4p4qgrojkj7pxo6iptx6epjqzfua32ufxyi.on.fleek.co/).

### CODE REVIEW

File: content/concepts/nat/dcutr.md
Comment (MarcoPolo) [inline content/concepts/nat/dcutr.md:13]: Can we please define a signaling server in this context?

File: content/concepts/nat/dcutr.md
Comment (MarcoPolo) [inline content/concepts/nat/dcutr.md:30]: The RTT is somewhat technical and also not really related to this paragraph?

File: content/concepts/nat/dcutr.md
Comment (MarcoPolo) [inline content/concepts/nat/dcutr.md:35]: This part is also somewhat technical, and I feel like would be better to just link to the spec directly.

**Review by MarcoPolo — COMMENTED** [review]
I think this page would be better as a much higher level overview of dcutr and nat holepunching. It can even be shorter. And simply link to the spec to have it explain the technical details.

**Comment by MarcoPolo** [comment]
Maybe we want to link: https://tailscale.com/blog/how-nat-traversal-works/

**Comment by marten-seemann** [comment]
> Maybe we want to link: https://tailscale.com/blog/how-nat-traversal-works/

Let's do that! At some point, I'd like to have our own post about NAT traversal that uses pretty diagrams, but for now, the Tailscale blog post is a good reference.

File: content/concepts/nat/dcutr.md
Comment (DannyS03) [reply content/concepts/nat/dcutr.md:35]: I think is important to have but simplified it even more. We also link to the spec at the bottom.

File: content/concepts/nat/dcutr.md
Comment (p-shahi) [inline content/concepts/nat/dcutr.md:9]:
```suggestion
Relays are used to traverse NATs by acting as proxies, but this can be expensive to scale and
```

File: content/concepts/nat/dcutr.md
Comment (p-shahi) [inline content/concepts/nat/dcutr.md:18]: How about we move this to the glossary  and link to it instead?

File: content/concepts/nat/dcutr.md
Comment (p-shahi) [inline content/concepts/nat/dcutr.md:13]:
```suggestion
maintain, and may result in low-bandwidth, high-latency connections. [Hole punching](hole-punching.md)
is another technique that enables NAT traversal by enabling two nodes behind NATs to communicate directly.
However, in addition to relay nodes, it requires another piece of infrastructure called signaling servers (for [rendezvous](../discovery/rendezvous.md) and synchronization).
```

File: content/concepts/nat/dcutr.md
Comment (p-shahi) [inline content/concepts/nat/dcutr.md:19]:
```suggestion
The good news is that libp2p offers a hole punching solution which eliminates not only the need for signaling servers but also allows the use of relay nodes to scale (by only relying on relay nodes temporarily).

```

**Comment by p-shahi** [comment]
@MarcoPolo Can you take a look at this PR, it looks good to merge from my end

**Review by p-shahi — APPROVED** [review]

**Comment by p-shahi** [comment]
@MarcoPolo or @mxinden need a review from either of you

**Review by MarcoPolo — APPROVED** [review]

**Comment by MarcoPolo** [comment]
This is ready, will merge when ci/builds finish

**Pull request merged**
